script/dump-symbols.py

```diff
@@ -54,8 +54,8 @@ def main(destination):
 
 
 def get_names_from_gyp():
-  variables = electron_gyp()
-  return (variables['project_name%'], variables['product_name%'])
+  variables = get_electron_branding()
+  return (variables['project_name'], variables['product_name'])
```

Comment:
Can we get `project_name` and `product_name` from the build configuration itself?
Like we do in `verify-ffmpeg.py`, https://github.com/electron/electron/blob/master/script/verify-ffmpeg.py#L24

```
product_name = gn(initial_app_path).args().get_string('electron_product_name')
```

Reply:
I'm roughly following how Chromium does things here. I'd rather avoid invoking `gn args` to get that info, because it's slow.

Can you expand on why you think the build args is the right place to put these variables? This solution was roughly informed by how Chromium/Google Chrome does it, see e.g. https://chromium.googlesource.com/chromium/src/+/67.0.3396.127/chrome/app/theme/chromium/BRANDING

Reply:
> I'd rather avoid invoking `gn args` to get that info, because it's slow.

Yes, it takes a few seconds, and it's bad because?.. )

> why you think the build args is the right place to put these variables? 

Because changing source files to customise a build looks weird.
There's literally no difference between changing the BRANDING.json and the BUILD.gn.
So how do we make Electron build more configurable by only extracting some data to a separate file?

I think that all types of customizations should be done in the same fashion, e.g. both turning a feature on or off, and setting an executable name. And gn args look like a perfect way to do it.

And since the gn configuration is "the source of truth", I think that we should not only set up a build via gn, but also retrieve information about a build via gn.